### ISSUE TITLE
E37: No write since last change (add ! to override)

### ISSUE THREAD

**Issue by sgebbie** [jakwings/vim-pony#4]
Hi,

I've run into a bug with `vim-pony`.

When I change a file and write the changes (`:w`), then if I wait more than 4 seconds (which seems to be the flush time to `.swp` and then try to quit (`:q`) I get the the error:

```
E37: No write since last change (add ! to override)
```

If I quit using `:wq` then the issue does not occur. But that is not my default work-flow.

I first assumed that this was a bug with a newer version of `vim`. But after some digging and eliminating `vim`, `vim-plug`, `YouCompleteMe` I worked out that it really was `vim-pony`. This surprised me because I wouldn't have thought that a syntax highlighting would have caused changes.

I eventually switched back to: [dleonard0/pony-vim-syntax](https://github.com/dleonard0/pony-vim-syntax) because it turned out I was still using that on another machine, and it confirmed my theory that the problem was with `vim-pony`.

My version of `vim` is the standard Ubuntu/Debian release version (in my digging I tried GTK and Gnome Vim etc.):

```
$ vim --version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:44:48)
Included patches: 1-1689
Extra patches: 8.0.0056
```

I don't have much knowledge of vim syntax highlighting, but if you have some pointers then maybe I can dig a bit deeper.

Thanks,
Stewart.

**Comment by iology** [comment]
It was caused by the automatic trailing space removal (noted in ReadMe), and should have been fixed by commit b26f01a869000b73b80dceabd725d91bfe175b75.

**Comment by sgebbie** [comment]
Awesome! Thanks for the quick resolution.
